### ISSUE TITLE
custom training class index problem

### ISSUE THREAD

**Issue by Mounthazeer** [ultralytics/yolov5#7430]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

hi, i am new to ml i just wanted some clarification
from what i have understood, we cannot add a new class to an existing weight file, we must train a new model

however if lets say i am training for 2 classes with my own dataset using pretrained weights where class index 0 is a new class that the pretrained  model was not trained on before and the class index 1 is an object that the pretrained model was already trained to detect but the class index was lets say 15

for example if i want my model to detect pencils and cars, now the class index is 0 for pencils and 1 for cars, however previously in the pretrained model it was already detecting cars with class index 2

is that going to be a problem?
is it okay to retrain the model this way such that now it is detecting only 2 classes and class 0 as pencil and class 1 as car, where previously it was detecting 80 classes with car in index 2

### Additional

in summary i want to train for 2 classes using pretrained weight but one class was already found in the coco dataset, from what i have understood i have to retrain the model again for the 2 classes, is this correct and can i still use pretrained weights for training?

**Comment by glenn-jocher** [comment]
@Mounthazeer see https://community.ultralytics.com/t/how-to-combine-weights-to-detect-from-multiple-datasets

**Comment by Mounthazeer** [comment]
thanks for the reply
but i do not want all the classes from coco i just need a specific one along with my own
so when training can i use *pretrained weights* to train a model with a class that was in coco and a new one with lets say with my own dataset

or should i start from scratch?

**Comment by glenn-jocher** [comment]
@Mounthazeer got it! Yes, you can absolutely use pretrained weights to train a model with a class from COCO dataset and another one from your own dataset. There is no need to start from scratch. You can utilize the pretrained weights and continue training the model with your specific classes. Good luck with your training!
